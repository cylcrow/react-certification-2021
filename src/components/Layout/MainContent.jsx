import React from 'react';
import styled from 'styled-components';

const StyledMainContent = styled.div`
  margin-top: 64px;
  height: 100%;
`;

const MainContent = () => {
  return <StyledMainContent data-testid="main-content">I&apos;m your main content</StyledMainContent>;
};

export default MainContent;
