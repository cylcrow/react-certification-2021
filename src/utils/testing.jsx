import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}