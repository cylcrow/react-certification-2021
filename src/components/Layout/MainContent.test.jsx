import React from 'react';
import '../../utils/testing';
import 'jest-styled-components';
import { render, unmountComponentAtNode } from 'react-dom';
import toJson from 'enzyme-to-json';
import MainContent from './MainContent';
import { mount } from 'enzyme';

let container;
const build = () => {
  render(<MainContent />, container);
};

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  return container;
});

describe("MainContent", () => {
  it('renders', () => {
    const wrapper = build(<MainContent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


describe("MainContent styles", () => {
  it("applies default styling", ()=>{
    const tree = mount(<MainContent />);
    expect(tree).toHaveStyleRule("margin-top", "64px");
  })
} );