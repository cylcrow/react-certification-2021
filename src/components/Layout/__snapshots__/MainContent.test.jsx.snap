// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent renders 1`] = `undefined`;
